app/src/views/Contribution.vue

```diff
@@ -0,0 +1,112 @@
+<template>
+  <template v-if="fullContributions && !loading">
+    <BaseHeader :name="title" />
+    <SectionHeader :title="`Contributions (${contributionTotal})`" />
+    <ContributionDetail :contributions="fullContributions" />
+  </template>
+  <LoadingSpinner v-else />
+</template>
+
+<script lang="ts">
+import BaseHeader from 'src/components/BaseHeader.vue';
+import SectionHeader from 'src/components/SectionHeader.vue';
+import ContributionDetail from 'src/components/ContributionDetail.vue';
+import useDataStore from 'src/store/data';
+import { computed, defineComponent, ref, watch } from 'vue';
+import LoadingSpinner from 'src/components/LoadingSpinner.vue';
+import { filterContributionGrantData } from 'src/utils/data/contributions';
+import useWalletStore from 'src/store/wallet';
+import { useRoute } from 'vue-router';
+import { GrantMetadata, GrantRoundMetadata } from '@dgrants/types';
+
+function setTitle(route: string) {
+  return route === '/contribution/donations' ? 'My Contributions' : 'Contributions';
+}
+
+function contributionDetails() {
+  const { grantContributions, grants, grantMetadata, grantRounds, grantRoundMetadata } = useDataStore();
+  const { userAddress } = useWalletStore();
+
+  const fullContributions = ref();
+  const contributionTotal = ref(0);
+  const loading = ref(true);
+
+  const allGrantRounds = computed(() => {
+    return grantRounds.value ? grantRounds.value : [];
+  });
+
+  const grantMetaData = computed(() => {
+    return grantMetadata.value as Record<string, GrantMetadata>;
+  });
+
+  const grantRoundsMetaData = computed(() => {
+    return grantRoundMetadata.value as Record<string, GrantRoundMetadata>;
+  });
+
+  const contributions = computed(() => {
+    return grantContributions.value ? grantContributions.value : [];
+  });
+
+  const allGrants = computed(() => {
+    return grants.value ? grants.value : [];
+  });
+
+  const userAddr = computed(() => {
+    return userAddress.value ? userAddress.value : '';
+  });
+
+  watch(
```

Comment:
Some comments on this component:
1. Seems all of these computed properties can be removed as they aren't doing anything significant except returning the same value, in some cases changing `undefined` to e.g. `[]`
2. Then, remove the watcher (there's no `async` calls in the watcher which typically means you don't need it) and can instead use computed properties which are lighter/more performant
3. `fullContributions` and `contributionTotal` should then become the two computed properties

Reply:
@mds1 I went ahead and updated this, I agree with what you are saying, should be resolved with latest push

Reply:
Thanks so much for this refactor, appreciate it!